### ISSUE TITLE
Add try/catch in dealloc

### ISSUE THREAD

**Issue by getaaron** [designatednerd/DNSCastroSegmentedControl#8]
It's possible for the control to be deallocated without registering for KVO. When this happens, this code will throw an exception:

```objc
- (void)dealloc
{
    [self removeObserver:self forKeyPath:[self boundsKeyPath]];
}
```

Here's the ugly solution as suggested by [NSHipster](http://nshipster.com/key-value-observing/):

```objc
- (void) dealloc {
    @try {
        [self removeObserver:self forKeyPath:[self boundsKeyPath]];
    } @catch(NSException * __unused exception){}
}
```

**Comment by designatednerd** [comment]
I'm super tempted to switch to MAKVO or some other similar implementation. KVOOOOOOOO!

**Comment by getaaron** [comment]
@designatednerd Come to think of it, I think we could just override `setBounds:`.

**Comment by designatednerd** [comment]
Fixed in 1.3.0
